reark/src/test/java/io/reark/reark/data/stores/cores/MemoryStoreCoreTest.java

```diff
@@ -148,6 +151,44 @@ public void delete_DoesNotTriggerStream() {
                 .assertNoValues();
     }
 
+    @Test
+    public void getCached_ReturnsAllValues() {
+        memoryStoreCore.put(100, "test value 1");
+        memoryStoreCore.put(200, "test value 2");
+
+        memoryStoreCore.getCached()
+                .test()
+                .assertValue(asList("test value 1", "test value 2"));
+    }
+
+    @Test
+    public void getCached_WhenEmptyStore_ReturnsEmptyList() {
+        memoryStoreCore.getCached()
+                .test()
+                .assertValue(emptyList());
```

Comment:
Just `.assertEmpty();`

Reply:
This is `Single`, so it returns an empty list in case of no items. Was that way since Rx1 didn't have `Maybe`. I think changing this to `Maybe` would make sense though

Reply:
@tomaszpolanski , the store API currently implements a null object pattern in case items are not found: e.g. `StoreGetInterface` has `getOnce` where the returned value is a `Single` with wrapper object, such as `Optional`. These could be changed to `Maybe`, but we still need the wrapping object to implement `getOnceAndStream` where we get initial null object in case of a missing item.

Perhaps changing those `Singles` to `Maybes` would make sense even if we need to keep the pattern for `Observables`?